core/src/main/java/org/keycloak/sdjwt/vp/SdJwtVP.java

```diff
@@ -104,8 +104,16 @@ public static SdJwtVP of(String sdJwtString) {
         int disclosureStart = sdJwtString.indexOf(SdJwt.DELIMITER);
         int disclosureEnd = sdJwtString.lastIndexOf(SdJwt.DELIMITER);
 
+        if (disclosureStart == -1) {
+            throw new IllegalArgumentException("SD-JWT is malformed, expected to contain a '" + SdJwt.DELIMITER + "'");
+        }
+
         String issuerSignedJWTString = sdJwtString.substring(0, disclosureStart);
-        String disclosuresString = sdJwtString.substring(disclosureStart + 1, disclosureEnd);
+        String disclosuresString ="";
```

Comment:
Better with spaces around the equal sign.
```suggestion
        String disclosuresString = "";
```